### ISSUE TITLE
Making new resource in the UI should not always use name in subject

### ISSUE THREAD

**Issue by Polleps** [atomicdata-dev/atomic-server#711]
## Problem:
After a change in how new resources are created in databrowser it always prefers using a name or shortname when it has one while generating a subject instead of using a random string. This makes sense in a lot of cases but not when the name is defaulted to something like 'Untiled Folder' or 'Untitled Document' because it results in ugly subjects that end in 'Untitled-Folder-[random string]'
<img width="462" alt="image" src="https://github.com/atomicdata-dev/atomic-server/assets/8804875/bcd99a53-1cf8-46ce-9ad9-9f8ae3443ea1">

## Possible Solutions:
1) Add an option to `useCreateAndNavigate` to always generate random string
2) Show name dialog when creating document or folder (Only fixes the problem for those two classes)
3) Always use UUID's i.e. `[driveSubject]/[UUID]` - `https://atomicdata.dev/51437179-c07c-4e6d-a94e-23cdae4d6f30`

**Comment by joepio** [comment]
Yeah I think the UUID strat is probably the way to go. Seems relatively easy to implement and prevents extremely long URLs.

**Comment by joepio** [comment]
A user asked this: 

> One bit I can't figure out is how to set the path when creating a new document in the UI.  It always uses "Untitled Document" and appends a random string to make it unique.
